sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -57,6 +57,11 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog with QueryErrorsB
   val DATA_TYPE_MISMATCH_ERROR = TreeNodeTag[Unit]("dataTypeMismatchError")
   val INVALID_FORMAT_ERROR = TreeNodeTag[Unit]("invalidFormatError")
 
+  // Error that has a lower priority, that are not supposed to throw immediately on triggering,
+  // e.g. certain internal errors. These errors will be thrown at the end of the whole check
+  // analysis process, if no other error occurs.
+  var preemptedError: Option[SparkException] = None
```

Comment:
Shall we make this a priority queue?  If there are more such errors, developers can set the priorities.

Reply:
I generalize it with a new class that records the 'heap' top. I only keep the top element since at most one error that has the highest priority gets thrown eventually. PTAL!